lib/model/model.go

```diff
@@ -252,24 +254,41 @@ func NewModel(cfg config.Wrapper, id protocol.DeviceID, clientName, clientVersio
 }
 
 func (m *model) serve(ctx context.Context) error {
-	// Add and start folders
+	m.initFolders()
+	m.cfg.Subscribe(m)
+	close(m.started)
+
+	var err error
+	select {
+	case <-ctx.Done():
+	case err = <-m.fatalChan:
+	}
+
+	m.cfg.Unsubscribe(m)
+	m.closeAllConnectionsAndWait()
```

Comment:
Defer these and return err/nil directly from the select, which becomes last in the method? I think this would be clearer than having err just be implicitly nil because it's unset, and keeps the unsubscribe with the subscribe.